File: src/main/java/com/manning/simplysend/services/OrdersService.java
package com.manning.simplysend.services;

import org.springframework.data.domain.Page;

import com.manning.simplysend.dto.OrderDTO;

public interface OrdersService {

    Page<OrderDTO> listOrders(String status, Integer page, Integer limit);

    void createOrder(OrderDTO orderDTO, String userEmailId);

    boolean approveOrder(Long orderID);

    boolean rejectOrder(Long orderID);
}

File: src/main/kotlin/com/manning/simplysend/services/OrdersService.kt
package com.manning.simplysend.services

import com.manning.simplysend.dto.OrderDTO
import org.springframework.data.domain.Page

interface OrdersService {

    fun listOrders(status: String?, page: Int, limit: Int): Page<OrderDTO>

    fun createOrder(orderDTO: OrderDTO, userEmailId: String)

    fun approveOrder(orderID: Long): Boolean

    fun rejectOrder(orderID: Long): Boolean
}